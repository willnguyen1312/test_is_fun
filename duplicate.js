export const getDuplicated = (array) => {
    const counting = {}
    const result = new Set()
    for (const item of array) {
        if (!counting[item]) {
            counting[item] = 1
        } else {
            result.add(item)
        }
    }

    return result;
}