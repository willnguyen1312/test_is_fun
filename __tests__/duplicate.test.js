import { getDuplicated } from "../duplicate"

const arr = [1, 1, 1, 3, 2, 3, 2, 4, 1, 2]

describe("Duplicate", () => {
    it("Should return array of duplicated numbers", () => {
        // expect(getDuplicated(arr)).toEqual([3, 2, 1])
        expect(getDuplicated(arr)).toMatchSnapshot()
    })
})