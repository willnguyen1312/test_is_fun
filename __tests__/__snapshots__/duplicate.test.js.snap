// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Duplicate Should return array of duplicated numbers 1`] = `
Set {
  1,
  3,
  2,
}
`;
